### PR TITLE
Link to search

### PR DESCRIPTION
**Pull request by alanlanglois**
## Detailed purpose of the PR

Go to search screen with given search params when clicking a link in the end popin for Adaptive
[trello](https://trello.com/c/lD4ZJNr2/1682-go-to-search-screen-with-given-search-params-when-clicking-a-link-in-the-end-popin-for-adaptive)

### Testing strategy

- [ ] I ensured my code is tested: 
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

File: src/modules/uri.js [hunk 1]
```diff
@@ -40,7 +41,30 @@ export const buildUrlQueryParams = (params: QueryParams) =>
...
+    'theme',
+    'author',
+    'type',
+    'authorType',
+    'lpath',
+    'sort',
+    'goal',
+    'search',
+    'playlist'
+  ];
```
Comment (VincentCATILLON) [inline src/modules/uri.js:60]: this kind of data, is related to search responsibility, this function should only return query params, and another function should filter properties

File: src/modules/uri.test.js [hunk 2]
```diff
@@ -31,4 +31,19 @@ describe('Uri', () => {
       expect(buildUrlQueryParams(params)).toBe(expectedResult);
     });
   });
+
+  describe('get query params', () => {
```
Comment (VincentCATILLON) [inline src/modules/uri.test.js:35]: camel case to retrieve it easily please

File: src/redux/actions/ui/search.test.js [hunk 3]
```diff
@@ -18,6 +18,20 @@ describe('search', () => {
     });
   });
 
+  describe('edit query params', () => {
```
Comment (VincentCATILLON) [inline src/redux/actions/ui/search.test.js:21]: camel case

File: src/redux/actions/ui/search.js [hunk 4]
```diff
@@ -22,6 +29,11 @@ export const edit = (payload: string): Action => ({
   payload
 });
 
+export const editQueryPrams = (payload: QueryParams): Action => ({
+  type: EDIT_QUERY_PARAMS,
+  payload
+});
```
Comment (VincentCATILLON) [inline src/redux/actions/ui/search.js:35]: why do we need this action ? as it should be called and dispatched in payload of search

File: src/screens/level-end.js [hunk 5]
```diff
@@ -100,6 +101,11 @@ class LevelEndScreen extends React.PureComponent<Props, State> {
     this.props.navigation.navigate('PdfModal', pdfParams);
   };
 
+  handleFeedbackLinkPress = (url: string) => {
+    const params = getQueryParams(url);
```
Comment (VincentCATILLON) [inline src/screens/level-end.js:105]: type

File: src/screens/search.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ import {HEADER_BACKGROUND_COLOR} from '../navigator/navigation-options';
 import Search from '../containers/search';
 
 export type Params = {|
-  url: string
+  url: string // is this in order to manage param in search?
```
Comment (VincentCATILLON) [inline src/screens/search.js:14]: ?

File: src/screens/level-end.js [hunk 7]
```diff
@@ -100,6 +101,11 @@ class LevelEndScreen extends React.PureComponent<Props, State> {
...
+    this.props.navigation.navigate('Search', params);
```
Comment (VincentCATILLON) [inline src/screens/level-end.js:106]: params should be typed as `SearchParams`

Comment (Fausto95) [reply src/screens/level-end.js:106]: getQueryParams has return type, so `const params = ` here is type inferenced

File: src/modules/uri.js (hunk 1 above)
Comment (Fausto95) [reply src/modules/uri.js:60]: I mean, either ways work here..

File: src/screens/level-end.js (hunk 7 above)
Comment (VincentCATILLON) [reply src/screens/level-end.js:106]: Seen with you, react navigation doesn't make the link between the params and the screen params we defined

File: src/redux/reducers/ui/search.js [hunk 8]
```diff
@@ -17,15 +17,15 @@ export const initialState: State = {
 const reducer = (state: State = initialState, action: Action): State => {
   switch (action.type) {
     case EDIT: {
+      if (typeof action.payload === 'string') {
```
Comment (Fausto95) [inline src/redux/reducers/ui/search.js:20]: Know someday I'll blame myself for writing code like this.
Should probably consider payload as object always and have something like
```
payload: {
 value: 'mySearchValue',
 params: { skill: 'someSkill', ...}
}

File: src/utils/search.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    'playlist'
+  ];
+  // $FlowFixMe
+  return allowedParams.reduce((acc, allowedParam) => {
+    if (allowedParam in params && params[allowedParam] !== undefined) {
+      acc[allowedParam] = params[allowedParam];
+      return acc;
+    }
+    return acc;
+  }, {});
```
Comment (VincentCATILLON) [inline src/utils/search.js:37]: I know that `lodash` flow coverage is not as good as we could use, but, if this function has unit test, we could consider using https://gist.github.com/jfmengels/6b973b69c491375117dc#_pickprops-object

**Pull request merged**